core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -179,6 +179,11 @@ private[spark] object ClosureCleaner extends Logging {
       cleanTransitively: Boolean,
       accessedFields: Map[Class[_], Set[String]]): Unit = {
 
+    if (!isClosure(func.getClass)) {
+      logWarning("Expected a closure; got " + func.getClass.getName)
```

Comment:
Any reason not to make this an assertion? Isn't it simply invalid if we call this on something that is not a closure?